lib/connection.js

```diff
@@ -579,6 +651,56 @@ Connection.prototype.decr = function(key, amount) {
 };
 
 /**
+ * gets() - Get a value and CID on this connection
+ *
+ * @param {String} key - The key for the value to retrieve
+ * @param {Object} [opts] - Any additional options for this get
+ * @returns {Promise} - containing an array with the value and CID
+ */
+Connection.prototype.gets = function(key, opts) {
+    opts = opts || {};
+    // Do the gets
+    var deferred = misc.defer(key);
+    this.commandBuffer = this.commandBuffer.push(deferred);
+
+    this.write('gets ' + key);
+
+    return deferred.promise
+    // .timeout() // @todo add this as a setting
+        .spread(function(data, flag, length, cas) {
+            if (data) {
+                if (opts.compressed || (flag & FLAG_COMPRESSED)) {
+                    // @todo compressed data should still be able to utilize the
+                    //  flags to return data of same type set
+                    return misc.decompress(new Buffer(data, 'base64'))
+                        .then(function(d) {
+                            return [ d.toString(), cas ];
+                        })
+                        .catch(function(err) {
+                            if (err.toString().indexOf('Error: incorrect header check') > -1) {
+                                // Basically we get this OperationalError when we try to decompress a value
+                                // which was not previously compressed. We return null instead of bubbling
+                                // this error up the chain because we want to represent it as a cache miss
+                                // rather than an actual error. By looking like a cache miss like this, it
+                                // makes it so someone can turn on compression and it'll work fine, looking
+                                // like a cache miss. So in the usual workflow the client tries to set it
+                                // again which means the second time around the compressed version is set
+                                // as the client wanted/expected.
+                                return [ null ];
```

Comment:
You mentioned this in your PR that it's a judgment call as to whether to return an array or an object.

There's no precedent for this in this library so it could go either way but I totally agree with you here.

However, if you do return an array with 2 elements I think it best to return the same always, so I would change the above line here to be: `return [ null, null ];` for consistency here and below.

It doesn't matter too much, but by leaving it only returning one `null` it is essentially looks like: `return [ null, undefined ];` which seems a bit misleading.